src/composables/guards.js

```diff
@@ -56,11 +56,23 @@ function useFilteredGuard (guard, fn) {
       : null
 
   if (depth != null) {
-    const removeGuard = router.beforeEach((to, from, next) => {
-      return fn(to, from, depth) ? guard(to, from, next) : next()
-    })
+    const registerGuard = () => {
+      return router.beforeEach((to, from, next) => {
+        return fn(to, from, depth) ? guard(to, from, next) : next()
+      })
+    }
 
+    let removeGuard = registerGuard()
     onUnmounted(removeGuard)
+
+    onActivated(() => {
+      removeGuard = removeGuard || registerGuard()
+    })
+    onDeactivated(() => {
+      removeGuard()
+      removeGuard = null // reset removeGuard
+    })
```

Comment:
Couldn't this just be
```suggestion
    onDeactivated(removeGuard)
```

Reply:
I need to reset `removeGuard` to `null` to ensure that `onActivated` will re-execute `registerGuard`, and `removeGuard` will not disappear after being called.

Reply:
Ah right, I thought that could be changed too but then I thought `onActivated()` could be called while a guard is active, creating multiple active guards too.